veps/sig-compute/secure-execution.md

```diff
@@ -0,0 +1,59 @@
+# Overview
+Implement support for IBM Z Secure Execution VMs in kubevirt
+
+## Motivation
+To enable customers to deploy secure workloads on their IBM Z machines using kubevirt.
+
+## Goals
+To enable kubevirt to run Secure Execution VMs on s390x
+
+## Non Goals
+Changing how existing VMs run.
+
+## Definition of Users
+- IBM Z users wanting to secure their new VMs using Secure Execution
+- IBM Z users wanting to migrate existing Secure Execution VMs to kubevirt
+
+## User Stories
+(list of user stories this design aims to solve)
+
+## Repos
+https://github.com/kubevirt/kubevirt
+https://github.com/kubevirt/api
+
+# Design
+
+The feature should be behind a Feature Gate at first, similiar to AMD SEV.
+For implementing this feature, the api behaviour would need to be changed to allow for setting an empty `launchSecurity: {}`.
+
+For launching Secure Execution VMs, all devices need to have `iommu=on` set to allow them to be visible to the guest, as in Secure Execution Mode the hypervisor is prevented from accessing the guest memory. Otherwise Secure Execution is initiaded from the guest image during boot, not the hypervisor.
+
+To ensure that VMs are scheduled only on nodes supporting Secure Execution, node-labeller should add a label to these nodes. This can be done by parsing the output of `virsh domcapabilities`.
+When a Secure Execution VM is created, the SE label will then be added to the virt-launcher pod so that it will be scheduled on a node which has the feature enabled.
+
+## API Examples
+```
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  ...
+spec:
+  domain:
+    launchSecurity: {}
+  ...
+```
+
+## Scalability
+It should scale like normal VMs.
+The only restriction would be for Secure Execution VMs, as they ´can't run on IBM Z machines that they have not been encrypted with.
+To move a Secure Execution VM to a new IBM Z machine, it's public key would first need to be added to the keys used to encrypt the kernel+initramfs+parmfile.
```

Comment:
Does it mean that migration is not possible by default?

Reply:
Migration is possible, but is limited by what the Guest Supports. Though the limit here is physical machines. This can be a bit confusing since usually a Machine is split into multiple LPAR (Logical Partitions) which is the closest to bare metal you can get on s390x.
The best thing here is to advice users to enable their VMs to run on all physical Machines they have in the cluster that support Secure Execution.

Reply:
To clarify, i mean offline migration. Live migration is not possible due to the restrictions on accessing the guest memory.

Reply:
Ok, I was worried about the live migration. Can you please specify that NonMigratableCondition will be added to these VMs?

Reply:
I have added it to the design section.